src/OpenTelemetry.AutoInstrumentation/ContinuousProfiler/ContinuousProfilerProcessor.cs

```diff
@@ -86,13 +100,20 @@ private static void ActivityChanged(AsyncLocalValueChangedArgs<Activity?> sender
         NativeMethods.ContinuousProfilerSetNativeContext(0, 0, 0);
     }
 
-    private static void SampleReadingThread(BufferProcessor sampleExporter, TimeSpan exportInterval)
+    private void SampleReadingThread(BufferProcessor sampleExporter)
     {
         while (true)
         {
-            // TODO Graceful shutdown and Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
-            Thread.Sleep(exportInterval);
+            // TODO Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
+            var shutdownRequested = _shutdownTrigger.Wait(_exportInterval);
```

Comment:
What is the typical timeout, where OTLPexporters are waiting?

Reply:
Default timeout is [5s](https://github.com/open-telemetry/opentelemetry-dotnet/blob/77ef12327f720ca3defd0c9590c0197cceb5952e/src/OpenTelemetry/BatchExportProcessor.cs#L18).

Reply:
Adjusted the code in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3294/commits/0a3b315354cf12b2a50421747a62ee4cb390a0ee to wait for the same amount (5s) as when tracer/meter provider are being disposed.